### DISCUSSION TITLE
pairwise PERMDISP no F values

### DISCUSSION THREAD

**Discussion by bark9299** [vegandevs/vegan#680] [Q&A]
I am using the commands:
```
permdisp_logCPM_wu_body <- betadisper(wu_dist_logCPM_body_site, ps_lcpm_40k_df$body.site)
test_permdisp <- permutest(permdisp_logCPM_wu_body, pairwise = TRUE)
```
But the resulting table only give p-values for each pairwise comparison, but not F. Is there a way to see what the F value is for all these pairwise comparisions? 

Thanks

**Comment by jarioksa** [comment]
Strange thing is that when I use the same commands, I get both a table of _p_-values and a table of  _F_ statistics. Can you prepare a _reproducible_ example, where 'reproducible' means that we can reproduce the example. Using, say, data in **vegan** is a good choice so that you do not need to provide your own data (and I usually do not want to download alien data).
```r
> m <- with(dune.env, betadisper(vegdist(dune), Management))
> permutest(m, pairwise = TRUE)

Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df  Sum Sq  Mean Sq      F N.Perm Pr(>F)
Groups     3 0.13831 0.046104 1.9506    999   0.15
Residuals 16 0.37816 0.023635                     

Pairwise comparisons:
(Observed p-value below diagonal, permuted p-value above diagonal)
         BF       HF       NM    SF
BF          0.570000 0.058000 0.267
HF 0.593549          0.076000 0.351
NM 0.060191 0.064936          0.372
SF 0.281235 0.402333 0.365221      
```

**Comment by bark9299** [comment]
Hi @jarioksa,

What I meant was that I don't get a F statistic for every pairwise comparision. Using the dune data example that you showed, I would like to know the F statistic of HF x BF, HF x NM, HF x SF.... It only gives you the overall F statisitic for all groups. Thanks

**Comment by jarioksa** [comment]
Why? Pairwise tests use _t_-statistics. As a first trick, you may try _F_ <sub>1,p</sub> = _t_<sub>p</sub><sup>2</sup>. Nothing of this makes sense in permutation tests that do not use theoretical distributions, but are based on ... permutations. I suggest you consult your statistical text book, and prepare to explain to referees why you used _F_ statistics in pairwise tests instead of _t_ statistics.

**Reply by bark9299** [reply]
You are right I am not extremely versed on statistics but I was following the way Qiime2 outputs pairwise permdisp. Qiime reports the F statistic for every pairwise comparision. Are you saying Qiime2 is wrong?

![image](https://github.com/user-attachments/assets/1e39ff5c-1caf-4be8-b0dd-b39798073e4a)

**Reply by gavinsimpson** [reply]
As a F statistic for a single term (e.g. an F test for a linear model with a single covariate - try it with a `lm()` and look at the F and t stat in the output) is directly comparable to a t statistic as @jarioksa showed ($F_{1,p} = t_{p}^2$; i.e. an F stat with 1 and $p$ degrees of freedom is equal to the squared t statistic with $p$ degrees of freedom), we're obviously not saying Qiime2 is wrong or right. Just stating a statistical fact.

**Comment by gavinsimpson** [comment]
If you want them, the observed statistics (F for the omnibus test, t for each pairwise test) are available in the `statistic` component of the object returned by `permutest.betadisper()`:

```r
# load data
data(varespec)

# Hellinger distances between samples
dis <- vegdist(varespec, method = "hellinger")

# First 16 sites grazed, remaining 8 sites ungrazed
groups <- factor(c(rep(1,16), rep(2,8)), labels = c("grazed","ungrazed"))

# Calculate multivariate dispersions
mod <- betadisper(dis, groups)
mod

# Perform test
anova(mod)

# Permutation test for F
pmod <- permutest(mod, permutations = 99, pairwise = TRUE)

# extract the observed test statistics
pmod$statistic
```

which gives

```      
r$> pmod$statistic
        Overall (F) grazed-ungrazed (t) 
           5.759668            2.399931
```

**Reply by gavinsimpson** [reply]
Or perhaps more interestingly (using @jarioksa's example):

```r
dij <- vegdist(dune, method = "hellinger")
m <- with(dune.env, betadisper(dij, Management))
ptest <- permutest(m, pairwise = TRUE)
ptest$statistic
```

yielding

```
r$> ptest$statistic
Overall (F)   BF-HF (t)   BF-NM (t)   BF-SF (t)   HF-NM (t)   HF-SF (t)   NM-SF (t) 
  1.5728727  -0.2813427  -2.2134247  -1.0345128  -1.8319082  -0.7853948   1.0470113
```

**Reply by gavinsimpson** [reply]
And of course, if you *must*, you can take the square root of the absolute value of the $t$ statistic to get $F$ statistics, but I really don't see the point, and you loose the sign this way:

```r
> sqrt(abs(ptest$statistic[-1]))
BF-HF (t) BF-NM (t) BF-SF (t) HF-NM (t) HF-SF (t) NM-SF (t) 
0.5304175 1.4877583 1.0171100 1.3534800 0.8862251 1.0232357 
```

The sign is useful because it tells you which group in each pair has the larger dispersion.

**Reply by jarioksa** [reply]
Like written in scriptures, _F_-values are squares of _t_, and you can always calculate _F_ from _t_. In a way it is only a convention which one to use. I think the main reasons for using _t_ instead of _F_ is that _t_ has sign and _t_ is two-sided: you see if the comparison means negative or positive difference. The code in `permutest.betadisper` uses `2 * pt(-abs(T0), df)` to get  _P_ from the theoretical distribution (as there are two tails). _F_ is always positive and you only have information of difference, but not of the direction of the difference. Would I develop Qiime I would use _t_, but it is not wrong to use _F_, in particular as Qiime seems to base their _P_ values on permutations, or at least they are not the analytic values from the theoretical _F_ distribution, and there _t_ and _F_ give same results. Consult **R** `help(pf)` for the relationship between _F_ and _t_.
